### ISSUE TITLE
Publishing to Jitpack and MavenCentral

### ISSUE THREAD

**Issue by kibotu** [vanniktech/gradle-maven-publish-plugin#884]
Sorry to abuse issues to ask a question, but is there a way to disable signing for jitpack?

currently I have to add a check, but it feels not like the intended way to do it. any suggestions? e.g. https://github.com/kibotu/AndroidResourceExtensions/blob/master/AndroidResourceExtensions/build.gradle#L77-L99

```groovy

if (!System.getenv("JITPACK") == "true") {
    apply plugin: 'com.vanniktech.maven.publish'
    mavenPublishing {
        publishToMavenCentral(com.vanniktech.maven.publish.SonatypeHost.CENTRAL_PORTAL, true)
        signAllPublications()
    }
}

if (System.getenv("JITPACK") == "true") {
    apply plugin: 'maven-publish'
    afterEvaluate {
        publishing {
            publications {
                release(MavenPublication) {
                    from components.release
                    groupId = GROUP
                    artifactId = POM_ARTIFACT_ID
                    version = version
                }
            }
        }
    }
}
```

**Comment by gabrielittner** [comment]
Is it specifically signing that fails on jitpack?

**Comment by kibotu** [comment]
Yes, mainly because I don't have a way to provide credentials to jitpack

**Comment by gabrielittner** [comment]
I can see simple solutions:
1. You do what you're doing already but just limit the if check for the `signAllPublications`. That way you can still let this plugin configure the rest of the publishing for jitpack and don't to manually apply `maven-publish`
2. Instead of using `signAllPublications()` you use the `RELEASE_SIGNING_ENABLED` Gradle property. That would allow to have signing off by default and then you can explicitly set it when publishing to maven central, for example by adding `-PRELEASE_SIGNING_ENABLED=true` to the Gradle publish command.
